Lib/test/test_random.py

```diff
@@ -436,6 +437,10 @@ def test_setstate_middle_arg(self):
             self.gen.setstate((2, (1,)*624+(625,), None))
         with self.assertRaises((ValueError, OverflowError)):
             self.gen.setstate((2, (1,)*624+(-1,), None))
+        # Failed calls to setstate() should not have changed the state.
+        bits100 = self.gen.getrandbits(100)
```

Comment:
I don't think that the test is correct. Why not generating bits before calling setstate()?